### PR TITLE
feat: added custom knowledge to chatgpt call

### PR DESCRIPTION
**Pull request by matt-newhall**

### CODE REVIEW

File: src/app/pages/api/openai.ts [hunk 1]
```diff
@@ -4,15 +4,28 @@ import { OpenaiApiType } from '@/app/types/OpenaiApiType';
...
   // source: https://supabase.com/docs/guides/getting-started/openai/vector-search
   const temperamentPrimer =
     'You are a helpdesk assistant for WizzAir who loves to help users find the correct information for their questions from our FAQs.';
   const knowledgeBasePrimer =
     'If you are unsure and the answer is not explicitly written in the information provided, say "Sorry, I don\'t know how to help with that.';
   const legalityPrimer = 'You should not provide legal advice of any kind.';
+
   const userDummyMessage = 'This is a test message!';
 
+  const customKnowledge = props.knowledge
```
Comment (SimonLTheodo) [inline src/app/pages/api/openai.ts:23]: Can maybe add a type for these {role, content} objects?

File: src/app/pages/api/openai.ts [hunk 2]
```diff
@@ -35,7 +49,7 @@ const fetchData = async (): Promise<OpenaiApiType> => {
   return data as OpenaiApiType;
 };
 
-fetchData().then(
+getChatResponse({ knowledge: [] }).then(
   (response) => console.log(response)
   // console.log(response.choices[0].message.content)
```
Comment (SimonLTheodo) [inline src/app/pages/api/openai.ts:54]: Could we remove this comment?

Comment (matt-newhall) [reply src/app/pages/api/openai.ts:54]: can do for now yes! we could just return the message now but imo easier for debugging if we just return the object, not like it's needed atm anyway

File: src/app/pages/api/openai.ts (hunk 1 above)
Comment (matt-newhall) [reply src/app/pages/api/openai.ts:23]: good shout!

**Review by SimonLTheodo — APPROVED** [review]

**Pull request merged**